src/core/jit/store.rs

```diff
@@ -1,22 +1,55 @@
+use std::collections::HashMap;
+
 use crate::core::jit::model::FieldId;
 
+/// Path to the data in the store with info
+/// to resolve nested multiple data
+#[derive(Debug, Clone)]
+pub struct DataPath {
+    /// List of paths, where every entry contains info about specific
+    /// level ot multiple data
+    multiple_path: Vec<usize>,
+}
+
+impl DataPath {
+    /// Create default DataPath that resolved to single value
+    pub fn single() -> Self {
+        Self { multiple_path: Vec::new() }
+    }
+
+    /// Create new DataPath with specified additional entry
+    pub fn with_index(&self, index: usize) -> Self {
```

Comment:
```suggestion
    pub fn with_index(self, index: usize) -> Self {
```

We shouldn't clone, let the cloning be managed by the caller so that if its possible they can optimize it.

Reply:
moved clone to the caller